### PR TITLE
MGMT-8097: masters schedulable kube-api

### PR DESCRIPTION
**Pull request by slaviered**
# Assisted Pull Request

## Description

see https://docs.openshift.com/container-platform/4.9/nodes/nodes/nodes-nodes-working.html#nodes-nodes-working-master-schedulable_nodes-nodes-working for more details

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: config/crd/bases/extensions.hive.openshift.io_agentclusterinstalls.yaml [hunk 1]
```diff
@@ -224,6 +224,9 @@ spec:
                       TODO: Add other useful fields. apiVersion, kind, uid?'
                     type: string
                 type: object
+              mastersSchedulable:
+                description: Set to true to allow control plane nodes to be schedulable
```
Comment (osherdp) [inline config/crd/bases/extensions.hive.openshift.io_agentclusterinstalls.yaml:228]: Just to match the surrounding documentation, plus some extra information:
```suggestion
                description: mastersSchedulable indicates whether or not to allow control plane nodes to schedule pods. Default is false.
```

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
@slaviered how did you test it?

File: config/crd/bases/extensions.hive.openshift.io_agentclusterinstalls.yaml (hunk 1 above)
Comment (slaviered) [reply config/crd/bases/extensions.hive.openshift.io_agentclusterinstalls.yaml:228]: This line was taken from the official OpenShift page documenting this property

File: api/hiveextension/v1beta1/agentclusterinstall_types.go [hunk 2]
```diff
@@ -155,6 +155,10 @@ type AgentClusterInstallSpec struct {
 	// DiskEncryption is the configuration to enable/disable disk encryption for cluster nodes.
 	// +optional
 	DiskEncryption *DiskEncryption `json:"diskEncryption,omitempty"`
+
+	// Set to true to allow control plane nodes to be schedulable
+	// +optional
+	MastersSchedulable bool `json:"mastersSchedulable,omitempty"`
```
Comment (rollandf) [inline api/hiveextension/v1beta1/agentclusterinstall_types.go:161]: Maybe MastersSchedulable -> ControlPlaneSchedulable

We should avoid 'master' terminology

Comment (slaviered) [reply api/hiveextension/v1beta1/agentclusterinstall_types.go:161]: This is an official attribute of openshift and we should preserve that naming. see the link in the description. It describe this attribute and its use

File: config/crd/bases/extensions.hive.openshift.io_agentclusterinstalls.yaml (hunk 1 above)
Comment (osherdp) [reply config/crd/bases/extensions.hive.openshift.io_agentclusterinstalls.yaml:228]: even if this line is from the bible, it needs to match conventions in this file :wink:
at the very least, give it the same structure as other description fields of this file, and keep the wording

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by filanov** [comment]
/hold 
Need to talk about this change.

**Comment by filanov** [comment]
We will close this PR for now.
Every change that we are doing to kube-api will become automatically supported, until we have specific request from the users we will not merge it. 
cc: @romfreiman @avishayt

**Comment by slaviered** [comment]
close until further notice

**Comment by zaneb** [comment]
/reopen
We are getting requests for this now.

**Pull request closed without merging**